### PR TITLE
Split admin and regular users in system-admin. Allow delete user

### PR DESCRIPTION
**Pull request by whaught**
Issue #https://github.com/google/exposure-notifications-verification-server/issues/975

## Proposed Changes

* Creates a new users tab in system-admin
    * Keeps system-admin list and regular users separate
* Allow system admin to delete a user
* Searching and pagination

withsearch

**Release Note**

```release-note
Show regular users across realms in system-admin. Allow deletion.
```

### CODE REVIEW

File: pkg/controller/admin/superusers.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		// Requested form, stop processing.
+		if r.Method == http.MethodGet {
+			var user database.User
+			c.renderNewUser(ctx, w, &user)
+			return
+		}
+
+		var form FormData
+		err := controller.BindForm(w, r, &form)
+		email := project.TrimSpace(form.Email)
```
Comment (sethvargo) [inline pkg/controller/admin/superusers.go:78]: This trimming and validation should happen in the model via a `BeforeSave`, not the controller.

File: pkg/controller/admin/users.go [hunk 2]
```diff
@@ -15,125 +15,32 @@
...
 
-	"github.com/google/exposure-notifications-verification-server/internal/auth"
-	"github.com/google/exposure-notifications-verification-server/internal/project"
 	"github.com/google/exposure-notifications-verification-server/pkg/controller"
 	"github.com/google/exposure-notifications-verification-server/pkg/database"
 	"github.com/gorilla/mux"
 )
 
-// HandleUsersIndex renders the list of system admins.
+// HandleUsersIndex renders the list of all non-system-admin usders.
```
Comment (sethvargo) [inline pkg/controller/admin/users.go:25]:
```suggestion
// HandleUsersIndex renders the list of all non-system-admin users.
```

File: pkg/database/user.go [hunk 3]
```diff
@@ -223,6 +223,29 @@ func (u *User) Stats(db *Database, realmID uint, start, stop time.Time) ([]*User
 	return stats, nil
 }
 
+// DeleteUser deletes the user entry.
+func (db *Database) DeleteUser(u *User) error {
+	return db.db.Delete(u).Error
+}
+
+// ListUsers returns a list of all users sorted by name.
```
Comment (sethvargo) [inline pkg/database/user.go:231]: Maybe a warning about using this vs `realm.ListUsers`?

File: pkg/database/user.go [hunk 4]
```diff
@@ -223,6 +223,29 @@ func (u *User) Stats(db *Database, realmID uint, start, stop time.Time) ([]*User
...
+func (db *Database) ListUsers() ([]*User, error) {
+	var users []*User
+	if err := db.db.
+		Model(&User{}).
+		Where("admin IS FALSE").
```
Comment (sethvargo) [inline pkg/database/user.go:236]: Why exclude admins?

File: pkg/database/user.go [hunk 5]
```diff
@@ -223,6 +223,29 @@ func (u *User) Stats(db *Database, realmID uint, start, stop time.Time) ([]*User
...
+		Order("LOWER(name) ASC").
+		Find(&users).
```
Comment (sethvargo) [inline pkg/database/user.go:238]: Pagination?

**Review by sethvargo — CHANGES_REQUESTED** [review]
This is looking good. I'd like us to avoid introducing a new term though. "super user" should be "system admin" everywhere, including routing, to avoid confusion.

File: pkg/database/user.go (hunk 4 above)
Comment (whaught) [reply pkg/database/user.go:236]: There are two pages: Admin users and Users. The delete action deletes the user here, but the admin users delete button just de-privileges the user by removing the admin bit.
I'm considering merging things or moving actions to a sub-page together, but for now I'm trying to keep them separate

File: pkg/database/user.go (hunk 5 above)
Comment (whaught) [reply pkg/database/user.go:238]: That's coming next!

File: pkg/controller/admin/superusers.go (hunk 1 above)
Comment (whaught) [reply pkg/controller/admin/superusers.go:78]: That does also happen in BeforeSave, but FindUserByEmail needs the trim

File: pkg/database/user.go (hunk 5 above)
Comment (whaught) [reply pkg/database/user.go]: aaand there it is

File: pkg/controller/admin/superusers.go (hunk 1 above)
Comment (sethvargo) [reply pkg/controller/admin/superusers.go:78]: Why not trim in `FindUserByEmail` then?

**Pull request merged**